erlang/README.md

```diff
@@ -0,0 +1,170 @@
+Overview
+========
+
+This document will enable you to bake erlang on rumprun and launch
+it within qemu, while running in xen falls on the same lines.
+
+Patches
+=======
+
+The patches within the `patches/` sub-directory is primarily to get
+ssl working, although in future it'll host some optimizations as
+discussed in `Unoptimized Memory Allocation`_ subsection.
+
+Instructions
+============
+
+Run this `make` from a user account, *not the root account*. At the end of the
+installation, the Makefile will build two variants of the build. One is
+required for a host, so that the erlang code can be compiled on this
+host (.erl) to .beam. There is another one which can be baked for rumprun
+kernel.
+
+The build script requires `genisoimage` to create `erlang.iso` image.
+
+Run `make` from the erlang package directory:
+
+```
+cd rumprun-packages/erlang
+make
+```
+
+Examples
+========
+
+There are sample erlang applications (echoserver and helloworld) which can
+be built via the `make examples`. The details of the build are captured in
+the Makefile.
+
+```
+make examples
+```
+
+Next you'll need a baked image which can be run on qemu (or kvm or xen) and the
+makefile is equipped to generate that as well (see below).
+
+The following build command will generate an image for non-smp, while using
+`hw` as the platform.
+
+```
+make beam.hw.bin
+```
+
+The above can be either used in qemu or kvm. In case you
+want to use xen then the following commands can be used towards the same.
+
+> Note that the build commands in turn use *rumpbake* internally, so
+> be sure that you have built the appropriate platform for it
+> to work. Additionally rumpbake with xen_pv is used in the Makefile.
+
+```
+make beam.xen.bin
+```
+
+Now you've got a baked unikernel image (hw_virtio - see Makefile). In
+case you have alternate needs then either update Makefile or bake
+your image on the shell. See the example below as an example, where
+the target is x86_64 (change it appropriate to your requirement).
+
+```
+rumpbake hw_virtio beam.bin build/bin/x86_64-rumprun-netbsd/beam
+```
+
+Setup the networking as follows (if not done already).
+
+```
+sudo ip tuntap add tap0 mode tap
+sudo ip addr add 10.0.120.100/24 dev tap0
+sudo ip link set dev tap0 up
+```
```

Comment:
This might be better served with a link e.g. to the wiki, since there's nothing Erlang-specific here.